airflow/models.py

```diff
@@ -3822,6 +3827,26 @@ def get_dag(self):
         return self.dag
 
     @provide_session
+    def get_previous_dagrun(self, session=None):
+        """The previous DagRun, if there is one"""
+
+        return session.query(DagRun).filter(
+            DagRun.dag_id == self.dag_id,
+            DagRun.execution_date < self.execution_date
```

Comment:
While it is outside the scope of this PR, this is potentially a quite expensive operation on the DB.